frontend/src/themes/transformThemeData.ts

```diff
@@ -0,0 +1,29 @@
+import defaultTheme from "./hubTheme";
+
+export const transformThemeData = (data) => {
+  const { palette, ...restOfDefaultTheme } = defaultTheme;
+  const customTheme = {
+    ...restOfDefaultTheme,
+    palette: {
+      ...palette,
+      primary: {
+        main: data?.primary?.main,
+        light: data?.primary?.light,
+        extraLight: data?.primary?.extraLight,
+        contrastText: data?.primary?.contrastText,
+      },
+      secondary: {
+        main: data?.secondary?.main,
+        light: data?.secondary?.light,
+        extraLight: data?.secondary?.extraLight,
+        contrastText: data?.secondary?.contrastText,
+      },
+      background: {
+        default: data?.primary?.background_default,
+        paper: data?.primary?.background_paper,
```

Comment:
This is a bit confusing because secondary.background_default and secondary.background_paper is never used. Maybe it's more intuitive if we directly add those to HubTheme without a foreign key between?